backend/instruments/tests.py

```diff
@@ -12,69 +12,133 @@ def load_tests(loader, tests, ignore):
+    endpoint = f"/instrument/{uuid}"
+
     @classmethod
     def setUpTestData(cls) -> None:
         company = Organization.objects.create(name="Test company")
         model = Model.objects.create(name="Test model")
+        test_type = Type.objects.create(name="Test type")
+        test_variable = Variable.objects.create(name="Test variable")
         model.manufacturers.add(company)
+        model.types.add(test_type)
+        model.variables.add(test_variable)
         model.save()
-        Instrument.objects.create(
-            uuid="d8b717b8-16e7-476a-9f5e-95b2a93ddff6",
-            name="Test instrument",
-            model=model,
+        instrument = Instrument.objects.create(
+            uuid=cls.uuid, name="Test instrument", model=model
         )
+        instrument.owners.add(company)
 
     def setUp(self):
         self.client = Client()
 
     def test_html(self):
-        response = self.client.get(
-            "/instrument/d8b717b8-16e7-476a-9f5e-95b2a93ddff6.html"
-        )
+        response = self.client.get(f"{self.endpoint}.html")
+        response_decoded = response.content.decode("utf-8")
         self.assertEquals(response.status_code, 200)
+        test_strings = (
+            "Test instrument",
+            "PID",
+            "Owners",
+            "Manufacturers",
+            "Test company",
+            "Model name",
+            "Test model",
+            "Instrument types",
+            "Test type",
+            "Measured variables",
+            "Test variable",
+            "Edit",
+            "JSON",
+            "XML",
+        )
+        for test_string in test_strings:
+            self.assertInHTML(test_string, response_decoded)
 
     def test_xml(self):
-        response = self.client.get(
-            "/instrument/d8b717b8-16e7-476a-9f5e-95b2a93ddff6.xml"
-        )
+        response = self.client.get(f"{self.endpoint}.xml")
         self.assertEquals(response.status_code, 200)
+        expected_xml = """
+        <instrument>
+        <identifier identifierType="Handle">20.1000/5555</identifier>
+        <schemaVersion>1.0</schemaVersion>
+        <landingPage>http://testserver/instrument/d8b717b8-16e7-476a-9f5e-95b2a93ddff6.html</landingPage>
+        <name>Test instrument</name>
+        <owners>
+        <owner>
+        <ownerName>Test company</ownerName>
+        </owner>
+        </owners>
+        <manufacturers>
+        <manufacturer>
+        <manufacturerName>Test company</manufacturerName>
+        </manufacturer>
+        </manufacturers>
+        <model>
+        <modelName>Test model</modelName>
+        </model>
+        <instrumentTypes>
+        <instrumentType>
+        <instrumentTypeName>Test type</instrumentTypeName>
+        </instrumentType>
+        </instrumentTypes>
+        <measuredVariables>
+        <measuredVariable>Test variable</measuredVariable>
+        </measuredVariables>
+        </instrument>
+        """
```

Comment:
Voisi olla helpompi lukea, jos näitä sisentäisi.

Reply:
Tai lukisi fixtuuri-tiedostosta yms.

Reply:
Joo totta tää on vähän ruma, korjataan.